lib/kubernetes-deploy/kubeclient_builder/kube_config.rb

```diff
@@ -1,21 +1,12 @@
 # frozen_string_literal: true
 
-require 'googleauth'
 module KubernetesDeploy
   class KubeclientBuilder
     class KubeConfig < Kubeclient::Config
       def self.read(filename)
         parsed = YAML.safe_load(File.read(filename), [Date, Time])
         new(parsed, File.dirname(filename))
```

Comment:
This is almost a verbatim copy of the Kubeclient version. IIRC we put this here in order to ensure the created object would call our version of `fetch_user_options` instead of `Kubeclient::Config`'s. (`new(parsed, File.dirname(filename)) vs `Config.new(parsed, File.dirname(filename))`.

Now that we don't need that, I think we can get rid of this class entirely.

cc @dturn @KnVerey

Reply:
I punted on removing this because https://github.com/Shopify/kubernetes-deploy/pull/464/files would have to put it back again.

Reply:
Can we do this work upstream in Kubeclient?

Reply:
I removed the file entirely since #464 was closed.